worker_test.go

```diff
@@ -27,6 +29,34 @@ func TestProcessMultipleJobs(t *testing.T) {
 	wg.Wait()
 }
 
+func TestDispatcherStopsWorkers(t *testing.T) {
+	var wg sync.WaitGroup
+
+	dispatcher := worker.NewDispatcher(100, func(payload worker.Payload) {})
+
+	// Between 1 & 6 workers
+	rand.Seed(time.Now().Unix())
+	numWorkers := rand.Intn(5) + 1
```

Comment:
If you are going to randomise the test conditions, I'd suggest you also log the conditions to stdout incase we have intermittent failures that only show up when eg numWorkers = 1.